### DISCUSSION TITLE
Trigger a chirp

### DISCUSSION THREAD

**Discussion by JamieP83** [tsightler/ring-mqtt#582] [Support Q&A] [answered]
hey all,
firstly, thank you! amazing work and add-on 
easy to install and so many great features 👍 

i didnt see this asked anywhere,  so sorry if this has been gone over before.

I have the ring security system (gen2) with some door sensors, when the doors open the do a "chirp"
this is great but its also very un-configurable - which is annoying via the ring app.

is this something available somehow, so we can configurte what/when does a "chirp"

**Comment by tsightler** [comment]
What do you mean by "un-configurable"?  The Ring app does allow chirp sounds to be changed or configured to not chirp at all.  Are you wanting to be able to changes these settings based on time-of-day or something?  If so, then the answer is no, these settings aren't currently supported.

**Comment by JamieP83** [comment]
Sorry, to be clear, i want to be able to tigger when/what causes a chirp.
eg. ring app wont let me mute all chirps or silence them for 30min. 
you also cant specify to not have things chrip when in the "disarmed" state (very annoying)

i was wondering if the "chirp" was availabe somehow, so i can trigger it via a call and then i can decide via automations when they occur.

**Comment by lougreenwood** [comment]
I also came looking for this.

For example, I have chirps triggered for motion in my basement, however my office is also in the basement, so when I go downstairs to work chirps are triggered.

This is fine most of the time, except when I get up to work early, then the keypad in my bedroom is chirping away whilst my wife sleeps... 😞.

@tsightler Is this something which the Ring API doesn't support at all, or just not supported in ring-mqtt?

Thanks!

**Reply by tsightler** [reply]
Sorry, but I'm not 100% clear on the "something" that you want to do?  In general the answer is the same as in the FAQ, if you can do it in the Ring app, then the API supports it, if you can't, then the API probably doesn't support it.

So, for example, it would be theoretically possible to make ring-mqtt support reconfiguring chirps for each device, since you can do that in the app, and it would be possible toggle the enable/disable chirps option for the keypad, since the keypad has this option in the Ring app, but neither of these functions are currently supported by ring-mqtt and I don't have any near term plans to implement it (at least the second option would be quite easy, so maybe I could sneak it in the next update, which should hit the world any day now). 

On the other hand, if the question is whether there's a way to make the base station and/or keypad play chirps via an automation, then I'm not aware of as there's no way to do this from the Ring app.

In the interim, it seems like your issue could potentially be addressed with the volume control, which is supported on both the keypad and base station even now.  You indicate that it is the keypad that is chirping so you could easily emulate the in-app switch to enable/disable chirps on the keypad by creating a switch in HA and having it set the volume to 0 when off, and set the volume to some reasonable level when on.

**Reply by tsightler** [reply, marked as answer]
In v5.6.0 I implemented a toggle for keypads to enable/disable chirps (same functionality as in the Ring app), so perhaps this might help

**Reply by JamieP83** [reply]
Perfect! Thanks a lot
Will play around with that

**Reply by lougreenwood** [reply]
Thanks @tsightler!

> On the other hand, if the question is whether there's a way to make the base station and/or keypad play chirps via an automation, then I'm not aware of as there's no way to do this from the Ring app.

After posting my earlier reply I realised that I actually have 2 use cases in mind:

### Programatically enable/disable chirps

This is my example use case above, disable chirps if the bedroom is in a sleeping state), which you're right could also be handled by volume automation too, but I'm super happy to see you were kind enough to sneak in toggling chirps in the device instead! Thank you! 🙇)

### Programatically trigger chirps from automation.

Here my use case is that the ring keypads make for really handy ways to alert of some state related to security. For example my office basement often has the windows left open and I wander upstairs for a few hours. In this case I'd like the sensors to chirp a specific sound if there is motion downstairs (kind of like a soft alarm, I want to know of movement when it isn't expected, but I don't want to sound the full alarm).

I would have an automation in NodeRed or similar which triggers a chirp when any ring motion sensor detects movement when the office didn't have occupancy before (there's some more nuance to this, but you get the idea).

---

But with all that said, I understand that there are limits to what the API makes possible, also there are limits to your time & attention and this is open source - so I already owe you a debt of gratitude for the immense effort to make it possible to have this integration for Ring at all.

I just wanted to share the above to add some clarity and throw an idea into the mix. To be honest my long-term hope is that Matter continues to grow in market share and Ring eventually releases an update or base station which supports matter and we can start to use that for local control. I'd give up all of the bells & whistles for local control and knowledge that a state change really happened even if the internet is down!

Again, thanks for all the work! 💙

**Reply by lougreenwood** [reply]
Just one more thing to note which I don't think I made clear.

Allowing to configure chimes/chirps for devices would go along way to allowing me to hack my "programatic automation" use case. In this case I could disable chirps for my basement motion sensors when I am downstairs working (the rest of the hose doesn't need to be annoyed by my movement), but when I go upstairs I "sort arm" the system by turning chirps on for my motion sensors.

I couldn't quite reach that goal with the current volume or keypad chirp toggle since I still need to hear chirps for contact sensors at all times when the system is disarmed.

**Reply by tsightler** [reply]
> Programatically trigger chirps from automation.

Unfortunately, I don't see this one happening.  Chirps are not triggered programmatically today.  The chirp configuration is stored locally in the base station (and perhaps keypad) and are triggered based on local Z-wave events.  There doesn't appear to be any way to trigger a chirp via the API (I can't even find a test function to exploit, which is how we do this in a few cases).

Note that it is possible to do this with a Ring Chime, and that option already exists today in ring-mqtt, so you can make the chime chirp for things other than Ring camera events.  Right now you can't change the chime sound via an automation, but this would be fairly easy to implement for chime and is on the to-do list.

> Allowing to configure chimes/chirps for devices would go along way to allowing me to hack my "programatic automation" use case.

Discussed in https://github.com/tsightler/ring-mqtt/discussions/720 already.  It's at least possible, just a pain to implement with the current ring-mqtt architecture which is built on the idea that each device is independent.  That just isn't the case for chirps as the settings for each sensor are actually stored centrally as part of the virtual "security-panel" device (presented as the alarm control panel, in ring-mqtt) and commands to change the chime must also be directed to this device.  I have ideas on how to deal with this that wouldn't be too difficult to implement, I just need to find time to flush it out, but, realistically, I regret not having built ring-mqtt with a device hierarchy although, at the time, there were only three supported devices and I didn't really see the future state we are now in.

**Reply by tsightler** [reply]
Programmatic enabling/disabling of chirp tones is implemented in the dev branch.  If you are willing to give it a spin and report back, please do so in thread https://github.com/tsightler/ring-mqtt/discussions/720.  Thanks!
